checkstyle.xml

```diff
@@ -28,6 +28,10 @@
     <module name="AvoidStarImport"/>
     <module name="IllegalImport"/> <!-- defaults to sun.* packages -->
     <module name="RedundantImport"/>
+    <module name="IllegalImport">
+      <property name="regexp" value="true" />
+      <property name="illegalClasses" value="^(?!.*(spoon|jdt)).*internal.*$" />
```

Comment:
Could you provide some examples in a comment of what exactly you want to catch? I'm not sure to understand it in the context of the PR so I guess I won't understand it at all in few weeks :) 

The regexp is to define which imports are illegal? Or in which package to look for illegal classes? So for example, does it look in https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/support/compiler/jdt/JDTImportBuilder.java? Thanks!

Reply:
This change adds a check look at all imports and stops user from importing internal packages expect spoon and jdt ones.
The regex looks for imports containing `internal` and **not** containing  spoon or jdt.

Some examples are:
`import java.util; //fine
import spoon.pattern.internal.node.ForEachNode; // allowed because internal package but from spoon.
import org.eclipse.jdt.internal.compiler.batch.CompilationUnit; //allowed because internal package but from jdt.
import org.apache.commons.jexl2.internal.introspection; //rejected because import of internal package`
Internal packages from jdt are allowed because it's needed in some spoon classes.
Should i add a comment in the pom for clarification?

Reply:
Thanks for the explanation :) 

> The regex looks for imports containing internal and not containing spoon or jdt.

nice I had to check back the Pattern documentation to find that `?!` was for a: 
> (?!X) | X, via zero-width negative lookahead

didn't know about that :)

> Should i add a comment in the pom for clarification?

That would be really cool. At least one line such as the quote above and one example of allowed/rejected. Thanks